src/events/MessageEvents.ts

```diff
@@ -370,10 +375,26 @@ export const messageEvents = async (agent: ServiceAgent, config: ServerConfig) =
     if (outOfBandRecordId) {
       const outOfBandRecord = await agent.oob.findById(outOfBandRecordId)
       const parentConnectionId = outOfBandRecord?.getTag('parentConnectionId') as string | undefined
+      await agent.modules.userProfile.requestUserProfile({ connectionId: payload.connection.id })
       if (!parentConnectionId)
         await agent.modules.userProfile.sendUserProfile({ connectionId: payload.connection.id })
     }
   })
+  
+  agent.events.on(
+    ProfileEventTypes.ConnectionProfileUpdated,
+    async ({ payload }: ConnectionProfileUpdatedEvent) => {
+      config.logger.debug(`ConnectionProfileUpdatedEvent received. Connection id: ${payload.connection.id} 
+        Profile: ${JSON.stringify(payload.profile)}`)
+
+      const msg = new ProfileMessage({
+        connectionId: payload.connection.id,
+        preferredLanguage: payload.profile.preferredLanguage,
```

Comment:
I think we should also send other fields, namely `displayPicture`, `displayName` and `description`.